python/hail/dataset.py

```diff
@@ -1019,181 +886,162 @@ def annotate_variants_expr(self, expr):
-
-        :rtype: :py:class:`.VariantDataset`
-        """
-
-        if isinstance(expr, list):
-            expr = ','.join(expr)
-
-        if vds_key is None:
-            jvds = self._jvds.annotateVariantsKeyTable(keytable._jkt, expr)
-        else:
-            jvds = self._jvds.annotateVariantsKeyTable(keytable._jkt, vds_key, expr)
-
-        return VariantDataset(self.hc, jvds)
-
-    @handle_py4j
-    def annotate_variants_intervals(self, input, root, all=False):
-        """Annotate variants from an interval list file.
-
-        **Examples**
-
-        Consider the file, *data/exons.interval_list*, in
-        ``chromosome:start-end`` format:
-
-        .. code-block:: text
-
-            $ cat data/exons.interval_list
-            1:5122980-5123054
-            1:5531412-5531715
-            1:5600022-5601025
-            1:5610246-5610349
-
-        The following invocation produces a vds with a new variant annotation,
-        ``va.inExon``. The annotation ``va.inExon`` is ``true`` for every
-        variant included by ``exons.interval_list`` and false otherwise.
-
-        >>> vds_result = vds.annotate_variants_intervals('data/exons.interval_list', 'va.inExon')
-
-        Consider the tab-separated, five-column file *data/exons2.interval_list*:
-
-        .. code-block:: text
-
-            $ cat data/exons2.interval_list
-            1   5122980 5123054 + gene1
-            1   5531412 5531715 + gene1
-            1   5600022 5601025 - gene2
-            1   5610246 5610349 - gene2
-
-        This file maps from variant intervals to gene names. The following
-        invocation produces a vds with a new variant annotation ``va.gene``. The
-        annotation ``va.gene`` is set to the gene name occurring in the fifth
-        column and ``NA`` otherwise.
-
-        >>> vds_result = vds.annotate_variants_intervals('data/exons2.interval_list', 'va.gene')
+        ``expr`` has the following symbols in scope:
 
-        **Notes**
+          - ``va``: variant annotations
+          - ``table``: See note.
 
-        There are two formats for interval list files.  The first appears as
-        ``chromosome:start-end`` as in the first example.  This format will
-        annotate variants with a *Boolean*, which is ``true`` if that variant is
-        found in any interval specified in the file and `false` otherwise.
+        .. note:: 
+        
+            The value of ``table`` inside root/expr depends on the number of values in the key table, 
+            as well as the ``product`` argument. There are three behaviors based on the number of values
+            and one branch for ``product`` being true and false, for a total of six modes:
+            
+            +-------------------------+-------------+--------------------+-----------------------------------------------+
+            | Number of value columns | ``product`` | Type of  ``table`` | Value of  ``table``                           |
```

Comment:
This table scrolls horizontally which isn't ideal.

Yeah, this is tough.  I don't see a better way, esp. with Bool => Int.